plugins/outputs/mongodb/mongodb_test.go

```diff
@@ -1,22 +1,53 @@
 package mongodb
 
 import (
+	"context"
+	"fmt"
+	"path/filepath"
 	"testing"
 	"time"
 
 	"github.com/influxdata/telegraf/config"
 	"github.com/influxdata/telegraf/plugins/common/tls"
 	"github.com/influxdata/telegraf/testutil"
 	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
 )
 
 func TestConnectAndWriteIntegrationNoAuth(t *testing.T) {
 	if testing.Short() {
 		t.Skip("Skipping integration test in short mode")
 	}
 
+	req := testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:        "mongo",
+			ExposedPorts: []string{"27017/tcp"},
+			WaitingFor:   wait.NewHTTPStrategy("/").WithPort("27017"),
+		},
+		Started: true,
+	}
+
+	ctx := context.Background()
+	container, err := testcontainers.GenericContainer(ctx, req)
+	require.NoError(t, err, "starting container failed")
+	defer func() {
+		require.NoError(t, container.Terminate(ctx), "terminating container failed")
+	}()
+
+	host, err := container.Host(ctx)
+	require.NoError(t, err, "getting container host address failed")
+	require.NotEmpty(t, host)
+
+	natPort, err := container.MappedPort(ctx, "27017/tcp")
+	require.NoError(t, err, "getting container host port failed")
+	port := natPort.Port()
+	require.NotEmpty(t, port)
```

Comment:
The testcontainers setup code is repeated three places with the only difference that I noticed being the ContainerRequest struct. It might be worth factoring these into a function that takes a ContainerRequest.